ts/components/messages/MessagesDeadlines.tsx

```diff
@@ -76,6 +77,48 @@ type State = {
   maybeLastLoadedStartOfMonthTime: Option<number>;
   lastMessagesState?: pot.Pot<ReadonlyArray<MessageState>, string>;
   allMessageIdsState: Set<string>;
+  isContinuosScrollEnabled: boolean;
+  lastDeadlineId: Option<string>;
+  nextDeadlineId: Option<string>;
+};
+
+/**
+ * Get the last deadline id (the oldest in time is the first in array position)
+ */
+export const getLastDeadlineId = (sections: Sections): Option<string> => {
+  return fromNullable(sections)
+    .chain(s => fpIndex(0, s))
+    .chain(d => fpIndex(0, [...d.data]))
+    .fold(none, item => {
+      if (!isFakeItem(item)) {
+        return some(item.e1.id);
+      }
+      return none;
+    });
+};
+
+/**
+ * Get the next deadline id
+ */
+export const getNextDeadlineId = (sections: Sections): Option<string> => {
```

Comment:
This function could be rewritten as follows:

```typescript
export const getNextDeadlineId = (sections: Sections): Option<string> => {
  const now = new Date().getTime();
  return sections
    .reduce<Option<MessageAgendaItem>>((acc, curr) => {
      const item = curr.data[0];
      // if item is fake, return the accumulator
      if (isFakeItem(item)) {
        return acc;
      }
      const newDate = new Date(item.e1.content.due_date).getTime();
      const diff = newDate - now;
      // if the acc is none, we don't need to make comparison with previous value
      if (isNone(acc)) {
        // just check the newDate is about future
        return diff >= 0 ? some(item) : none;
      }
      const lastDate = acc.value.e1.content.due_date.getTime();
      // if the new date is about future and is less than in accomulator
      if (diff >= 0 && diff < now - lastDate) {
        return some(item);
      }
      return acc;
    }, none)
    .map(item => item.e1.id);
};
```

If you are agree about this implementation, it would be great if you write some tests on it